BookingProject/BookingWebAPI/BookingWebApi.Application/Apartament/IApartamentRepository.cs

```diff
@@ -22,5 +22,6 @@ public interface IApartamentRepository
         Task<Result<List<TotalAreaCountBySourceDto>>> GetTotalAreaCountBySourceCompany();
         Task<Result<AreaQuantilesDto>> GetAreaQuantiles();
         Task<Result<bool>> UpsertCustomData<T>(int apartamentId, T customData);
+        Task<Result<(decimal TotalPrice, string CurrencyName)>> CalculateTotalPriceWithCurrency(int apartamentId, DateTime startDate, DateTime endDate);
```

Comment:
Don't use tuples they are not as good extendable as classes and records. Create a small model for it. For me it looks like value object pattern for TotalPrice and CurrencyName